### PR TITLE
feat: Adding Examples for Python SDK to create FCR and FCR based Connections

### PR DESCRIPTION
**Pull request by jkallem-equinix**
CXF-96059 Adding Examples for Python SDK:
- Fabric Cloud Router
        1. Create FCR 
        2. GET FCR
        3. Update FCR
        4. Delete FCR

FCR to Port, AWS, Azure Connections:
- Connections
      1.  Create Connection 
      2. GET Connection
      3. Update Connection
      4. Delete Connection

### CODE REVIEW

File: examples/services/fabricv4/cloud_router/cloud_router.py [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+    Create a Fabric Cloud Router (FCR) using the Equinix Fabric API.
+
+    This method sends a request to create a new Fabric Cloud Router with specified configurations 
+    such as router type, name, location, package, and associated project and account details. 
+    Notifications are also configured to receive updates regarding the router.
+
+    Construct a `CloudRouterPostRequest` object using the `module.CloudRouterPostRequest` class.
+    This object contains the details required to create the Fabric Cloud Router.
+
+    Notes:
```
Comment (thogarty) [inline examples/services/fabricv4/cloud_router/cloud_router.py:18]: Remove the notes sections of all comments.

File: examples/services/fabricv4/cloud_router/cloud_router.py [hunk 2]
```diff
@@ -0,0 +1,93 @@
+from examples.services.fabricv4.utils import utils
+from equinix.services import fabricv4 as module
```
Comment (thogarty) [inline examples/services/fabricv4/cloud_router/cloud_router.py:2]: Can we import just as fabricv4 and not as module? Better for code readability to have a specific name other than an abstract one.

File: examples/services/fabricv4/cloud_router/cloud_router_management.py [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+"""
+    Create a Fabric Cloud Router (FCR) using the provided request payload.
+
+    Args:
+        fcr_request: An instance of the request payload object for creating a Fabric Cloud Router.
+
+    Returns:
+        str: The UUID of the created Fabric Cloud Router.
+
+    Steps:
```
Comment (thogarty) [inline examples/services/fabricv4/cloud_router/cloud_router_management.py:19]: You don't need to add steps in comments. They can read the code for this.

File: examples/services/fabricv4/cloud_router/cloud_router_management.py [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+"""
+    Retrieve the details of an existing Fabric Cloud Router (FCR) by its UUID.
+
+    Args:
+        fcr_uuid (str): The UUID of the Fabric Cloud Router to retrieve.
+
+    Returns:
+        None: This method prints the details of the FCR to the console.
+
+    Steps:
```
Comment (thogarty) [inline examples/services/fabricv4/cloud_router/cloud_router_management.py:53]: All steps from comments can be removed.

File: examples/services/fabricv4/cloud_router/cloud_router_management.py [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+    Update an existing Fabric Cloud Router (FCR) using the provided UUID and update payload.
+
+    Args:
+        fcr_uuid (str): The UUID of the Fabric Cloud Router to update.
+        update_payload: An instance of the update payload object containing the updates for the FCR.
+
+    Returns:
+        None: This method prints the update response to the console.
+
+    Steps:
```
Comment (thogarty) [inline examples/services/fabricv4/cloud_router/cloud_router_management.py:82]: I'm not going to comment on all steps removals but please go through and remove all references to notes and steps in comment strings.

File: examples/services/fabricv4/connections/connection_management.py [hunk 6]
```diff
@@ -0,0 +1,115 @@
+import json
+import time
+
+
+from icecream import ic
+from examples.services.fabricv4.utils import utils
+from examples.services.fabricv4.oauth2 import configure_client_credentials
+from examples.services import fabricv4
```
Comment (thogarty) [inline examples/services/fabricv4/connections/connection_management.py:8]: There's an import collision here. Why are both being imported?

File: examples/services/fabricv4/connections/connection_management.py [hunk 7]
```diff
@@ -0,0 +1,115 @@
...
+
+        This method handles the process of establishing a connection between a Fabric Cloud Router (FCR) and a
+        colocation or another FCR within the Equinix Fabric. It generates the connection request payload,
+        sends the request to the Equinix Fabric API, and returns the unique identifier (UUID) of the created connection.
+
+        Returns:
+            str: The UUID of the newly created connection, which can be used for further operations like querying the connection
+                 status, updating, or deleting the connection.
+
+        Example Usage:
```
Comment (thogarty) [inline examples/services/fabricv4/connections/connection_management.py:24]: Please remove example usage from comment strings.

File: examples/services/fabricv4/connections/fcr_2_aws_connection.py [hunk 8]
```diff
@@ -0,0 +1,180 @@
+import time
+import os
+
+from examples.services.fabricv4.utils import utils
+from equinix.services import fabricv4 as module
```
Comment (thogarty) [inline examples/services/fabricv4/connections/fcr_2_aws_connection.py:5]: Check all imports and don't rename to module.

File: examples/services/fabricv4/connections/connection_management.py (hunk 6 above)
Comment (thogarty) [reply examples/services/fabricv4/connections/connection_management.py:8]: I think that `from examples.services import fabricv4` should be removable.

File: examples/services/fabricv4/connections/fcr_2_azure_connection.py [hunk 9]
```diff
@@ -0,0 +1,141 @@
+import time
+import os
+
+from examples.services.fabricv4.utils import utils
+from equinix.services import fabricv4 as module
```
Comment (thogarty) [inline examples/services/fabricv4/connections/fcr_2_azure_connection.py:5]: Don't need the as in import.

**Review by thogarty — CHANGES_REQUESTED** [review]
Some changes requested. Please review comments as well as this summary:

1. The comment strings are very distracting. Please reduce them as much as possible. No steps, notes, example usages. Give a very short summary of what is about to happen and then let the user read the code for better understanding. We can't maintain this level of commenting. Good code speaks for itself and doesn't need to be overly explained.
2. The imports need to be cleaned up. Don't import anything `as module` because it loses the context of its namespace and becomes incredibly confusing.
3. The custom_serializer in utils, I couldn't find where it was used because of all of the distractions. Is it necessary?

File: examples/services/fabricv4/connections/fcr_2_azure_connection.py (hunk 9 above)
Comment (displague) [reply examples/services/fabricv4/connections/fcr_2_azure_connection.py:5]: I think we should add `black` to the GH actions to reduce the need for style feedback in reviews.

Similar action in another Equinix SDK: https://github.com/packethost/packet-python/blob/master/.github/workflows/test.yml#L15-L26

**Pull request closed without merging**